### PR TITLE
Fixes exception handling stack traces

### PR DESCRIPTION
**Pull request by goatslacker**
On some cases we pass/throw a non-error, we used to wrap that in an
Error and then serialize that new error. The problem with that approach
is that the stack trace would be completely lost.

Now we either include the original stack trace or no stack trace at all.
Because an incorrect stack trace is worse than no stack trace.

@ljharb @lencioni

### CODE REVIEW

File: src/utils/BatchManager.js [hunk 1]
```diff
@@ -2,8 +2,13 @@ function errorToSerializable(error) {
   // istanbul ignore next
   if (error === undefined) throw new TypeError('No error was passed');
 
-  // make sure it is an error object so we can serialize it properly
-  const err = error instanceof Error ? error : new Error(error);
+  // make sure it is an object that is Error-like so we can serialize it properly
+  // if it's not an actual error then we won't create an Error so that there is no stack trace
+  // because no stack trace is better than a stack trace that is generated here.
+  const err = typeof error === 'object' && typeof error.stack === 'string'
```
Comment (lencioni) [inline src/utils/BatchManager.js]: Why did you change from the `instanceof` check you were previously using?

Comment (goatslacker) [inline src/utils/BatchManager.js]: because errors might come from different places such as VMs and `instanceof` might not work

Comment (ljharb) [inline src/utils/BatchManager.js]: `instanceof` doesn't work across realms, like `vm` - you probably want `Object.prototype.toString.call(error) === '[object Error]' && typeof error.stack === 'string'` so as to be a bit more robust.

File: src/utils/BatchManager.js [hunk 2]
```diff
@@ -2,8 +2,13 @@ function errorToSerializable(error) {
...
+    ? error
```
Comment (lencioni) [inline src/utils/BatchManager.js]: Is it useful to include this in the stack trace? Perhaps this should be `? new Error(error.stack)`?

Comment (goatslacker) [inline src/utils/BatchManager.js]: That would nest the error messages/stack trace. We don't actually need this to be an Error. Error-like is ok.

File: src/utils/BatchManager.js (hunk 1 above)
Comment (goatslacker) [inline src/utils/BatchManager.js]: I kinda like the loose requirements now, means that you can pass in a POJO with a custom stack. Although I'm not sure what you'd use this for.

Comment (ljharb) [inline src/utils/BatchManager.js]: If you really want that, you could set `Symbol.toStringTag` to `"Error"` - but I don't think it's a good idea to allow pojos. Error stacks will be standardized sooner than later, and I'd think we want proper stack traces.
